R/type-factor.R

```diff
@@ -180,10 +156,42 @@ vec_cast.ordered <- function(x, to, ...) {
   UseMethod("vec_cast.ordered")
 }
 #' @export
+#' @method vec_cast.ordered ordered
+vec_cast.ordered.ordered <- function(x, to, ...) {
+  fct_cast(x, to, ...)
+}
+#' @export
 #' @method vec_cast.ordered factor
 vec_cast.ordered.factor <- function(x, to, ...) {
   fct_cast(x, to, ...)
 }
+#' @export
+#' @method vec_cast.ordered character
+vec_cast.ordered.character <-function(x, to, ...) {
+  fct_cast(x, to, ...)
+}
+#' @export
+#' @method vec_cast.character ordered
+vec_cast.character.ordered <- function(x, to, ...) {
+  stop_native_implementation("vec_cast.character.ordered")
+}
+
+fct_cast <- function(x, to, ..., x_arg = "", to_arg = "") {
```

Comment:
I think the right place for this kind of function is just below the exported method-generic.

We might want to split this one into `fct_cast()` and `ord_cast()`. I think `fct_cast()` should mean "cast any subclasses of factors to factor". In the same way, `df_cast()` casts any subclasses of data frames to data frame.